src/protocol-parser/protocol-parser.ts

```diff
@@ -618,3 +625,291 @@ export namespace Session {
     return parseObject(params, SubscriptionRequestParametersSchema);
   }
 }
+
+/** @see https://w3c.github.io/webdriver-bidi/#module-input */
+export namespace Input {
+  // input.ElementOrigin = {
+  //   type: "element",
+  //   element: script.SharedReference
+  // }
+  const ElementOriginSchema = zod.object({
+    type: zod.literal('element'),
+    element: CommonDataTypes.SharedReferenceSchema,
+  });
+
+  // input.Origin = "viewport" / "pointer" / input.ElementOrigin
+  const OriginSchema = zod.union([
+    zod.literal('viewport'),
+    zod.literal('pointer'),
+    ElementOriginSchema,
+  ]);
+
+  // input.PauseAction = {
+  //   type: "pause",
+  //   ? duration: js-uint
+  // }
+  const PauseActionSchema = zod.object({
+    type: zod.literal(InputTypes.ActionType.Pause),
+    duration: zod.number().nonnegative().int().optional(),
+  });
+
+  // input.KeyDownAction = {
+  //   type: "keyDown",
+  //   value: text
+  // }
+  const KeyDownActionSchema = zod.object({
+    type: zod.literal(InputTypes.ActionType.KeyDown),
+    value: UnicodeCharacterSchema,
+  });
+
+  // input.KeyUpAction = {
+  //   type: "keyUp",
+  //   value: text
+  // }
+  const KeyUpActionSchema = zod.object({
+    type: zod.literal(InputTypes.ActionType.KeyUp),
+    value: UnicodeCharacterSchema,
+  });
+
+  // input.TiltProperties = (
+  //   ? tiltX: -90..90 .default 0,
+  //   ? tiltY: -90..90 .default 0,
+  // )
+  const TiltPropertiesSchema = zod.object({
+    tiltX: zod.number().min(-90).max(90).int().default(0).optional(),
+    tiltY: zod.number().min(-90).max(90).int().default(0).optional(),
+  });
+
+  // input.AngleProperties = (
+  //   ? altitudeAngle: float .default 0.0,
+  //   ? azimuthAngle: float .default 0.0,
+  // )
+  const AnglePropertiesSchema = zod.object({
+    altitudeAngle: zod
+      .number()
+      .min(0.0)
```

Comment:
Looks like the spec does not define min/max? should we accept any angle and do mod?

Reply:
It's defined. See step (29) of https://w3c.github.io/webdriver/#dfn-process-a-pointer-up-or-pointer-down-action